examples/mobile/ScheduleApp/js/app.js

```diff
@@ -22,42 +30,139 @@
+	prototype._onDateSelected = function (event) {
+		var calendarEl,
+			calendar;
+
+		if (event.target.id === "date-on") {
+			calendarEl = document.getElementById("calendar-on");
+		} else if (event.target.id === "date-off") {
+			calendarEl = document.getElementById("calendar-off");
+		}
+		if (calendarEl) {
+			calendar = tau.widget.Calendar(calendarEl);
+			calendar.value(event.detail.datetime);
+			calendar.element.classList.remove("app-hidden");
+			event.target.classList.add("app-hidden");
+		}
+	};
+
+	// this feature should belong part of TAU
+	prototype._onCalendarSwitch = function (event) {
+		var dateEl,
+			date;
+
+		if (event.target.id === "calendar-on") {
+			dateEl = document.getElementById("date-on");
+		} else if (event.target.id === "calendar-off") {
+			dateEl = document.getElementById("date-off");
+		}
+		if (dateEl) {
+			date = tau.widget.DateTimePicker(dateEl);
+			date.value(event.detail.date);
+			date.element.classList.remove("app-hidden");
+			event.target.classList.add("app-hidden");
+		}
+	};
+
+	prototype._beforeShowCreateEvent = function (event) {
+		var page = event.target,
+			ui = this._ui;
+
+		ui.doneButton = page.querySelector(".app-event-done");
+		ui.dateOn = page.querySelector("#date-on");
+		ui.dateOff = page.querySelector("#date-off");
+		ui.dayOfWeek = page.querySelector(".ui-day-of-week-picker");
+
+		ui.doneButton.addEventListener("vclick", this, false);
+
+		page.addEventListener("selected", this, false);
+		page.addEventListener("calendarswitch", this, false);
+	};
+
+	prototype._beforeShowMain = function (event) {
 		var page = event.target;
 
-		if (page.id === "create-event-page") {
-			page.addEventListener("selected", function (event) {
-				var calendarEl,
-					calendar;
+		console.log("_initMain", page);
+		// read data from storage
+		// fill page content
+		// refresh page
+	};
 
-				if (event.target.id === "date-on") {
-					calendarEl = document.getElementById("calendar-on");
-				} else if (event.target.id === "date-off") {
-					calendarEl = document.getElementById("calendar-off");
-				}
-				if (calendarEl) {
-					calendar = tau.widget.Calendar(calendarEl);
-					calendar.value(event.detail.datetime);
-					calendar.element.classList.remove("app-hidden");
-					event.target.classList.add("app-hidden");
-				}
-			});
-			page.addEventListener("calendarswitch", function (event) {
-				var dateEl,
-					date;
-
-				if (event.target.id === "calendar-on") {
-					dateEl = document.getElementById("date-on");
-				} else if (event.target.id === "calendar-off") {
-					dateEl = document.getElementById("date-off");
-				}
-				if (dateEl) {
-					date = tau.widget.DateTimePicker(dateEl);
-					date.value(event.detail.date);
-					date.element.classList.remove("app-hidden");
-					event.target.classList.add("app-hidden");
-				}
-			});
+	prototype._eventDoneClick = function () {
+		var eventData = {},
+			self = this,
+			ui = self._ui,
+			router;
```

Comment:
you define ui as
ui = sefl._ui
so define store as
store = self._store

Reply:
Local variables are created for properties used more than once. But this is sample app and we don't need these rule because this app probably will not be minified.